### ISSUE TITLE
metadata_options default not working properly in Launch Template

### ISSUE THREAD

**Issue by phils** [hashicorp/terraform-provider-aws#12564]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.24
+ provider.aws v2.55.0

### Affected Resource(s)

* aws_launch_template
(maybe others using metadata options too)

### Terraform Configuration Files

```hcl
resource "aws_launch_template" "lt" {
  name                   = "test-lt"
  image_id               = var.ami
  instance_type          = var.instance_type
  user_data              = base64encode(data.template_file.userdata.rendered)
  key_name               = var.key_name
  vpc_security_group_ids = [aws_security_group.blah.id]

  metadata_options {
    http_tokens = "required"
  }
}


```

### Expected Behavior

LT with metadata options as requested

### Actual Behavior

Error: InvalidParameterValue: A value of ‘’ is not valid for http-endpoint. Valid values are ‘enabled’ or ‘disabled’.

This seems to be a common issue with launch template actions in terraform AWS Provider.  When leaving items to 'default', I believe you must omit the variable altogether in the AWS API call, rather than sending blank string value.

### Steps to Reproduce

metadata_options where http_endpoint is not specified in the tf code.

### References
https://github.com/terraform-providers/terraform-provider-aws/pull/12491

* #0000

**Comment by phils** [comment]
@ewbankkit FYI - appreciate your work on this code but might need a small tweak

**Comment by ewbankkit** [comment]
@phils Thanks for opening this issue.
The new `metadata_options` attribute is defined as `Computed: true` for backwards compatibility reasons: https://github.com/terraform-providers/terraform-provider-aws/pull/11076#discussion_r378475598.

**Comment by stefansundin** [comment]
@phils 

In this case you must specify `http_endpoint = "enabled"`. In my opinion that attribute should be `Required: true`, but it is not currently.

```hcl
resource "aws_launch_template" "lt" {
  name                   = "test-lt"
  image_id               = var.ami
  instance_type          = var.instance_type
  user_data              = base64encode(data.template_file.userdata.rendered)
  key_name               = var.key_name
  vpc_security_group_ids = [aws_security_group.blah.id]

  metadata_options {
    http_endpoint = "enabled"
    http_tokens   = "required"
  }
}
```

**Comment by bharath-123** [comment]
I think the documentation should be updated in this case. https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/launch_template#metadata-options says http_endpoint is optional which is misleading.

**Comment by justinretzolk** [comment]
Hey @phils :wave: Thank you for taking the time to file this. Given that there's been a number of AWS Provider releases since you initially filed it, can you confirm whether you're still experiencing this behavior?

**Comment by RyPeck** [comment]
@justinretzolk I just experienced this behavior creating a `aws_instance` Resource Type using the `v3.62.0` AWS Provider.

**Comment by Dinoshauer** [comment]
Just ran into this for `aws_instance.metadata_options.http_endpoint` as well with `hashicorp/aws` at `4.15.0`

**Comment by kylet21** [comment]
My team also just ran into this issue. Can we at least get an update in the documentation that says, "`http_endpoint = "enabled"` must be set if you want to set any other options in the `metadata_options` block"?

**Comment by rodecapd** [comment]
> My team also just ran into this issue. Can we at least get an update in the documentation that says, "`http_endpoint = "enabled"` must be set if you want to set any other options in the `metadata_options` block"?

Same here using provider v4.47 and terraform-aws-modules/eks/aws v19.5.1.
I've been trying to set
metadata_options {
  http_token = "optional"
} 
and it didn't work until I added "`http_endpoint = "enabled"`

**Comment by lvthillo** [comment]
@ewbankkit 

> @phils Thanks for opening this issue. The new `metadata_options` attribute is defined as `Computed: true` for backwards compatibility reasons: [#11076 (comment)](https://github.com/hashicorp/terraform-provider-aws/pull/11076#discussion_r378475598), which is why we have no `Default` value here.


I'm only seeing this now (after PR creation)... Not sure if this backwards compatibility is still required?

**Comment by justinretzolk** [comment]
Hey all 👋 Thank you very much for taking the time to raise this! This was addressed with #30545, which was included in version `5.0.0` of the provider. With that in mind, we'll close this issue. If you experience additional issues with the provider, please do open a new issue to let us know.
